esp-hal/src/dma/gdma.rs

```diff
@@ -297,122 +243,177 @@ impl<const N: u8> RegisterAccess for Channel<N> {
             .modify(|_, w| w.inlink_start().set_bit());
     }
 
-    fn is_in_done() -> bool {
-        Self::in_int().raw().read().in_suc_eof().bit()
-    }
-
-    fn is_listening_in_eof() -> bool {
-        Self::in_int().ena().read().in_suc_eof().bit_is_set()
-    }
-
-    fn is_listening_out_eof() -> bool {
-        Self::out_int().ena().read().out_total_eof().bit_is_set()
-    }
-
-    fn listen_in_eof() {
-        Self::in_int().ena().modify(|_, w| w.in_suc_eof().set_bit());
-    }
-
-    fn listen_out_eof() {
-        Self::out_int()
-            .ena()
-            .modify(|_, w| w.out_total_eof().set_bit());
-    }
-
-    fn unlisten_in_eof() {
-        Self::in_int()
-            .ena()
-            .modify(|_, w| w.in_suc_eof().clear_bit());
-    }
-
-    fn unlisten_out_eof() {
-        Self::out_int()
-            .ena()
-            .modify(|_, w| w.out_total_eof().clear_bit());
-    }
-
-    fn listen_ch_in_done() {
-        Self::in_int().ena().modify(|_, w| w.in_done().set_bit())
-    }
-
-    fn clear_ch_in_done() {
-        Self::in_int()
-            .clr()
-            .write(|w| w.in_done().clear_bit_by_one());
-    }
-
-    fn is_ch_in_done_set() -> bool {
-        Self::in_int().raw().read().in_done().bit()
+    fn listen_out(interrupts: EnumSet<DmaTxInterrupt>) {
```

Comment:
If these can be `impl Into<EnumSet<...>>` I think you can avoid the countless `enum_set!` macros in the callers.

Reply:
Oh nice, I'll give that a shot